### PR TITLE
(feat) Add pyarrow extras env markers

### PR DESCRIPTION
**Pull request by kukushking**
### Relates
- #1518 
- #1731

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,7 +30,10 @@ boto3 = "^1.24.11"
 botocore = "^1.27.11"
 pandas = "^1.2.0, <=1.5.1, !=1.5.0" # Exclusion per: https://github.com/aws/aws-sdk-pandas/issues/1678
 numpy = "^1.22.4"
-pyarrow = ">=6.0.0, <8.1.0"
+pyarrow = [
+    { version = ">=6.0.1, <8.0.0", markers = "extra == 'ray' or 'ray' in extra" },
```
Comment (kukushking) [inline pyproject.toml:34]: Even though `in` is defined by the [grammar](https://peps.python.org/pep-0508/#complete-grammar) I couldn't quite make it work for the combinations of extras.. Just putting it out there

Comment (LeonLuttenberger) [reply pyproject.toml:34]: Is this saying that we should only install `pyarrow` when we use `ray`? Don't we use `pyarrow` in non-distributed `read_parquet` as well?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: e193484392adce28968ea9e874ac50d8ff9ee212
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1624385c-7da3-46f5-8712-87317dea4d8d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: e193484392adce28968ea9e874ac50d8ff9ee212
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=61395bf1-9b0f-4af4-8320-4757a8f61c73%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: pyproject.toml (hunk 1 above)
Comment (kukushking) [reply pyproject.toml:34]: There's 2nd line below that that covers non-ray

Comment (LeonLuttenberger) [reply pyproject.toml:34]: 🤦 my bad!

It looks like the static check build is still failing with:
> ModuleNotFoundError: No module named 'pyarrow'

**Comment by kukushking** [comment]
@jaidisido yeah I agree if GA is far from here no point in having this. I'm ok to close this for now but let's have a chat about dates first

**Comment by kukushking** [comment]
Closing this for now as agreed. We will revise this closer to the GA.

**Pull request closed without merging**